test/scheduler_tests.cc

```diff
@@ -0,0 +1,230 @@
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+#include <catch2/matchers/catch_matchers.hpp>
+
+#include "scheduler_test_utils.h"
+#include "test_utils.h"
+
+
+using namespace celerity;
+using namespace celerity::detail;
+namespace acc = celerity::access;
+
```

Comment:
Maybe I missed it, but is there a test checking whether commands, which should cause a flush, do so?